README.md

```diff
@@ -7,32 +7,136 @@ Inspired from [Caddy](https://github.com/mholt/caddy).
+
+#### Modularity
+
+Maddy does have a module-based design and you specify in configuration modules
+you want to use and how you want them to work.
+
+Generic syntax for configuration block is as follows:
 ```
-imaps://127.0.0.1:1993 {
-	proxy imaps://mail.example.org
-	tls self_signed
-	compress
-	pgp
+module_name instance_name {
+    configuration_directives
 }
+```
+
+`instance_name` is the unique name of the configuration block. It is used when
+you need to refer to the module from different place in configuration (e.g.
+configure SMTP to deliver mail to certain specific storage)
+
+You can omit `instance_name`. If there is only one module config. block - it
+will get name the same as `module_name`. If there are multiple config blocks
+for one module - they will get names `module_name`, `module_name1`,
+`module_name2` and so on.
+
+#### Defaults
+
+Maddy provides reasonable defaults so you can start using it without spending
+hours writing configuration files. All you need it so define smtp and imap
+modules in your configuration, configure TLS (see below) and set domain name.
 
-smtps://127.0.0.1:1025 {
-	proxy smtps://mail.example.org
-	tls self_signed
-	pgp
+Here is the minimal example to get you started:
+```
+imap imap://0.0.0.0 imaps://0.0.0.0 {
+    tls cert_file pkey_file
+}
+
+smtp smtp://0.0.0.0 smtps://0.0.0.0:587 {
+    tls cert_file pkey_file
+    hostname emersion.fr
 }
 ```
+Don't forget to use actual values instead of placeholders.
+
+With this configuration, maddy will create an SQLite3 database for messages in
+current directory and use it to store all messages.
+
+#### go-sqlmail: Database location
+
+If you don't like SQLite3 or don't want to have it in the current directory,
+you can override the configuration of the default module.
+
+See [go-sqlmail repository](https://github.com/foxcpp/go-sqlmail) for
+information on RDBMS support.
+
+```
+sqlmail default {
+    driver sqlite3
+    dsn file_path
+}
+```
+
+You can then replace SQL driver and DSN values. Note that maddy needs to be
+built with a build tag corresponding to the name of the used driver (`mysql`,
+`postgresql`) for SQL engines other than sqlite3.
+
+DSN is a driver-specific value that describes the database to use.
+For SQLite3 this is just a file path.
+For MySQL: https://github.com/go-sql-driver/mysql#dsn-data-source-name
+For PostgreSQL: https://godoc.org/github.com/lib/pq#hdr-Connection_String_Parameters
+
+Note that you can also change default DSN or SQL driver during compilation
+by building maddy using following command:
+```shell
+go build -ldflags "-X github,com/emersion/maddy.defaultDriver=DRIVER -X github.com/emersion/maddy.defaultDsn=DSN"
+```
+
+#### TLS
+
+Currently, maddy doesn't implement any form of automatic TLS like Caddy. But
+since we don't want to have insecure defaults we require users to either
+manually configure TLS or disable it explicitly.
+
+Valid variants of TLS config directive:
+
+Disable TLS:
+```
+tls off
```

Comment:
Maybe add a comment `# this is insecure` just to make sure people don't copy-pasta

Reply:
Ok.